spec/rails_helper.rb

```diff
@@ -52,3 +52,12 @@
   # https://relishapp.com/rspec/rspec-rails/docs
   config.infer_spec_type_from_file_location!
 end
+
+#####DELETES FILES IN SPECIFIED FOLDER SO TEST FILES AREN'T PERSISTING
+RSpec.configure do |config|
+  config.after(:each) do
+    if Rails.env.test?
+      FileUtils.rm_rf(Dir["#{Rails.root}/public/uploads/user/avatar/*"])
```

Comment:
I would consider instead saving test attachments to a different path in your uploader. 

in your uploader you can do something like:

```ruby
def store_dir
   if Rails.env.test?
    File.join(Rails.root.join('tmp'), super))
  else
    super
  end
end
```